scripts/docker/buildpack-deps/Dockerfile.ubuntu.clang.ossfuzz

```diff
@@ -59,26 +62,19 @@ RUN apt-get update; \
 		pygments-lexer-solidity \
 		pylint \
 		requests \
-		tabulate \
-		z3-solver;
```

Comment:
Z3 is already installed from source in the steps below, so this is unnecessary. I also removed CMake, as it is already included in the base image.

Reply:
This is the python package of `Z3`, which is unrelated to the Z3 binary built below.
However, if it can be removed without anything breaking, that it's good to remove it :)

Reply:
Yeah, it is not used by any CI job based on the ossfuzz image as far as I know. The [chk_proofs](https://github.com/ethereum/solidity/blob/71988f7546063a9d6d63d54ece85839400b46264/.circleci/config.yml#L943) which is the job that runs the script mentioned here: https://github.com/ethereum/solidity/pull/15551#discussion_r1821295419 uses the Ubuntu 24.04 image and it is not run by ossfuzz. This is why I'm also wondering if all the smt stuff is really needed here, or if we should just remove it from this image.

Reply:
Ok, I updated Z3 and CVC5, but I kept the removal of the `z3-python`, I don't think it is relevant for this image. Because it is required by the tests under `test/formal`, which indeed uses the z3 python bindings, but I believe the fuzzer does not run such tests. Maybe there are more things that we could cleanup as well.